Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2098,22 +2099,40 @@ void CGContextShowGlyphsWithAdvances(CGContextRef context, const CGGlyph* glyphs
 #pragma endregion
 
 #pragma region Drawing Operations - Basic Shapes
+
+HRESULT __CGContext::ClearRect(CGRect rect) {
+    PushBeginDraw();
+    auto endDraw = wil::ScopeExit([this]() { PopEndDraw(); });
+
+    ComPtr<ID2D1RectangleGeometry> rectangle;
+    RETURN_IF_FAILED(Factory()->CreateRectangleGeometry(__CGRectToD2D_F(rect), &rectangle));
+
+    // Transformed geometry
+    ComPtr<ID2D1TransformedGeometry> transformedRectangle;
+    RETURN_IF_FAILED(Factory()->CreateTransformedGeometry(rectangle.Get(),
+                                                          __CGAffineTransformToD2D_F(CGContextGetUserSpaceToDeviceSpaceTransform(this)),
+                                                          &transformedRectangle));
+
+    RETURN_IF_FAILED(PushGState());
+
+    RETURN_IF_FAILED(CurrentGState().IntersectClippingGeometry(transformedRectangle.Get(), kCGPathEOFill));
+
+    ComPtr<ID2D1SolidColorBrush> brush;
+    RETURN_IF_FAILED(deviceContext->CreateSolidColorBrush(D2D1::ColorF(D2D1::ColorF::Black, 0.0f), &brush));
+
+    deviceContext->SetPrimitiveBlend(D2D1_PRIMITIVE_BLEND_COPY);
+    deviceContext->FillGeometry(CurrentGState().clippingGeometry.Get(), brush.Get());
```

Comment:
I wonder how this works in combination with transparency layers. Can you do a test on our platform and the reference platform where you draw red, begin a transparency layer, draw blue, clear a sub-region, and end the layer? I'd expect the blue to have a "hole" in it where the red is visible.